vehicle.go

```diff
@@ -0,0 +1,164 @@
+package smartcar
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/smartcar/go-sdk/helpers/requests"
+	"github.com/smartcar/go-sdk/helpers/utils"
+)
+
+// Vehicle contains vehicle request information such as ID and AccessToken and used for Smartcar vehicle functions.
+type Vehicle struct {
+	ID          string
+	AccessToken string
+	UnitSystem  string
+}
+
+// VehicleInfoResponse contains the vehicle information response returned from the VehicleInfo function.
+type VehicleInfoResponse struct {
+	ID    string `json:"id"`
+	Make  string `json:"make"`
+	Model string `json:"model"`
+	Year  int    `json:"year"`
+}
+
+// VehicleResponse contains a general vehicle status response returned from the VehicleLock, VehicleUnlock and VehicleDisconnect functions.
+type VehicleResponse struct {
+	Status string `json:"status"`
+}
+
+// SetUnits takes a vehicle and sets the unit system that information for the vehicle will be returned in.
+func (vehicle *Vehicle) SetUnits(unit string) error {
+	if !(unit == "metric" || unit == "imperial") {
+		return errors.New("unit must either be metric or imperial")
+	}
+	vehicle.UnitSystem = unit
+	return nil
+}
+
+// Info sends request to Smartcar API vehicle/ endpoint and returns formatted response.
+func (vehicle *Vehicle) Info() (VehicleInfoResponse, error) {
+	res, err := vehicle.request("/", requests.GET, nil)
+	if err != nil {
+		return VehicleInfoResponse{}, err
+	}
+
+	formattedResponse := new(VehicleInfoResponse)
+	defer res.Body.Close()
+	fmtErr := requests.FormatResponse(res.Body, formattedResponse)
+	if fmtErr != nil {
+		return VehicleInfoResponse{}, fmtErr
+	}
+
+	return *formattedResponse, nil
+}
+
+// VIN sends request to Smartcar API vehicle/vin endpoint and returns formatted response.
+func (vehicle *Vehicle) VIN() (string, error) {
+	res, resErr := vehicle.request("/vin", requests.GET, nil)
+	if resErr != nil {
```

Comment:
Typically I've seen just `err` unless there is a particular reason to name them with the extra info (e.g. the error is used somewhere other than the proceeding line, so a longer name would make it easier to keep track of the variable)